src/commands/ping/ping.ts

```diff
@@ -1,24 +1,43 @@
 import { Command, container } from '@sapphire/framework';
 import { Message } from 'discord.js';
-import { CodeyCommand, SapphireMessageExecuteType, SapphireMessageResponse } from '../../codeyCommand';
+import {
+  CodeyCommand,
+  SapphireMessageExecuteType,
+  SapphireMessageRequest,
+  SapphireMessageResponse
+} from '../../codeyCommand';
 
-const executeCommand: SapphireMessageExecuteType = (
+const initialPingContent = 'Ping?';
+
+const getApiLatency = (initialPing: SapphireMessageRequest, messageFromUserAsMessage: Message<boolean>) => {
+  const initialPingAsMessage = <Message<boolean>>initialPing;
+  return initialPingAsMessage.createdTimestamp - messageFromUserAsMessage.createdTimestamp;
+};
+
+const content = (botLatency: number, apiLatency: number) =>
+  `Pong from JavaScript! Bot Latency ${botLatency}ms. API Latency ${apiLatency}ms.`;
+
+const executeCommand: SapphireMessageExecuteType = async (
   client,
   messageFromUser,
   initialMessageFromBot
 ): Promise<SapphireMessageResponse> => {
   // Assert message types are Message<boolean>
   // We have to do this because APIMessage does not have "createdTimestamp" property
   const messageFromUserAsMessage = <Message<boolean>>messageFromUser;
-  const initialMessageFromBotAsMessage = <Message<boolean>>initialMessageFromBot;
   const botLatency = client.ws.ping;
-  const apiLatency = initialMessageFromBotAsMessage.createdTimestamp - messageFromUserAsMessage.createdTimestamp;
-  const content = `Pong from JavaScript! Bot Latency ${botLatency}ms. API Latency ${apiLatency}ms.`;
-  return new Promise((resolve, _reject) => resolve(content));
+  if (initialMessageFromBot) {
+    const apiLatency = getApiLatency(initialMessageFromBot, messageFromUserAsMessage);
+    return new Promise((resolve, _reject) => resolve(content(botLatency, apiLatency)));
+  }
+  const initialPing = await messageFromUserAsMessage.channel.send(initialPingContent);
+  const apiLatency = getApiLatency(initialPing, messageFromUserAsMessage);
+  initialPing.edit(content(botLatency, apiLatency));
+  return Promise.resolve('');
```

Comment:
if we always return smth, will we ever get fail msg?

Reply:
maybe i just dont get it 😓

Reply:
Yup, if any line of code before the return does not work, then the failure message will send.